### DISCUSSION TITLE
Error while compiling in esphome

### DISCUSSION THREAD

**Discussion by sergiolar** [syssi/esphome-atorch-dl24#44] [General]
Hi, 
Your work is excellent !!!

I want to capture the data coming from my AT3010 unit in Home Assistant via ESP-WROOM-32 with its adapted code.

I am compiling using the ESPHOME dashboard your code adapting only names to the ipaddress and it gives me the following errors (they are at the end)

Power BT unit:    **AT3010**
ESP32 board:     **ESP-WROOM-32**

**Adapted code:**

[esp32-at3010.yaml.txt](https://github.com/syssi/esphome-atorch-dl24/files/10094550/esp32-at3010.yaml.txt)

**compilation / installation**

```
INFO Reading configuration /home/homeassistant/.homeassistant/esphome/esp32-at3010.yaml...
INFO Updating https://github.com/syssi/esphome-atorch-dl24.git@main
INFO Generating C++ source...
INFO Compiling app...
Processing at3010-ac-meter (board: esp32dev; framework: arduino; platform: platformio/espressif32 @ 3.5.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
Dependency Graph
|-- WiFi @ 1.0
|-- ESPmDNS @ 1.0
|-- Update @ 1.0
Compiling .pioenvs/at3010-ac-meter/src/main.cpp.o
Linking .pioenvs/at3010-ac-meter/firmware.elf
RAM:   [==        ]  16.2% (used 53104 bytes from 327680 bytes)
Flash: [========  ]  78.0% (used 1431146 bytes from 1835008 bytes)
Building .pioenvs/at3010-ac-meter/firmware.bin
esp32_create_combined_bin([".pioenvs/at3010-ac-meter/firmware.bin"], [".pioenvs/at3010-ac-meter/firmware.elf"])
Wrote 0x16d6d0 bytes to file /home/homeassistant/.homeassistant/esphome/.esphome/build/at3010-ac-meter/.pioenvs/at3010-ac-meter/firmware-factory.bin, ready to flash to offset 0x0
======================== [SUCCESS] Took 898.20 seconds ========================
INFO Successfully compiled program.
INFO Connecting to 192.168.0.5
INFO Uploading /home/homeassistant/.homeassistant/esphome/.esphome/build/at3010-ac-meter/.pioenvs/at3010-ac-meter/firmware.bin (1431248 bytes)
Uploading: [============================================================] 100% Done...


INFO Waiting for result...
INFO OTA successful
INFO Successfully uploaded program.
**_ERROR No remote or local logging method configured (api/mqtt/logger)_**
```

Where am i going wrong?
I would appreciate your help
Thank you

**Comment by syssi** [comment]
It's important to use the `ESP-IDF` framework instead of `Arduino`:

```
esp32:
  board: esp32dev
  framework:
    type: esp-idf
    version: latest
```

Please add the `api:` component/line to your YAML so you are able to receive logs via WiFi after flashing. As soon you are able to compile and flash the new build please increase the log level to provide some debug output at the log:

```
logger:
  baud_rate: 0
  level: VERY_VERBOSE
  logs:
    api.service: WARN
    ota: WARN
    wifi: WARN
    sensor: DEBUG
```

**Reply by syssi** [reply]
Any news here?

**Reply by RenPio2001** [reply]
Hi,
Your work is excellent!!!
I have Atorch DT3010 and it works on ESPHome in Home Assistant, my settings:

esp32:
  board: esp32dev
  framework:
    type: arduino

logger:
  baud_rate: 0
  level: VERY_VERBOSE
  logs:
    api.service: WARN
    ota: WARN
    wifi: WARN
    sensor: DEBUG

Is it possible to connect a second Atorch under one ESP32, any tips?
Thanks

**Reply by syssi** [reply]
@RenPio2001 Could you create a new issue? I will provide an example how to talk to 3 devices using a single ESP32.
